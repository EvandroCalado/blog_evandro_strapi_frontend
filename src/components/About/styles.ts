import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    ${Title} {
      font-size: ${theme.font.sizes['4xl']};
    }
`}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.md};

    &::first-letter {
      font-size: ${theme.font.sizes.xl};
      font-weight: bold;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`

    & > img {
      width: 35rem;
      height: 35rem;
      overflow: hidden;
      border-radius: ${theme.spacings.md};
      object-fit: cover;
    }
`}
`;
