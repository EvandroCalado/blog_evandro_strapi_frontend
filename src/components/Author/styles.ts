import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    margin: ${theme.spacings.xl} auto;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
      width: 30rem;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 22rem;
    height: 17rem;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
    margin-right: ${theme.spacings.md};

    & img {
      object-fit: cover;
    }

    @media screen and ${theme.media.lteOrEqSmall} {
      width: 15rem;
      margin-bottom: ${theme.spacings.lg};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;

    @media screen and ${theme.media.lteOrEqSmall} {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.sm} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 3;
`}
`;
