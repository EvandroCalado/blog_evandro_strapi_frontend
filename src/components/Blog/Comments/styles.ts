import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;

    iframe {
      color: ${theme.colors.text};
    }
`}
`;
