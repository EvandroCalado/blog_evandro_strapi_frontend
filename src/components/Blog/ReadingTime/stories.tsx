import { Meta, StoryFn } from '@storybook/react';
import ReadingTime, { ReadingTimeProps } from '.';

import mock from '../../../utils/db.json';
const content = mock.data.posts.data[0].attributes;

export default {
  title: 'Components/Blog/ReadingTime',
  component: ReadingTime,
  args: content,
} as Meta;

export const Default: StoryFn<ReadingTimeProps> = (args) => (
  <ReadingTime {...args} />
);
