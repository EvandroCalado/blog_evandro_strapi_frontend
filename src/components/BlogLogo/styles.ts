import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Title = styled.h1`
  text-transform: uppercase;
  letter-spacing: 0.7rem;
  font-weight: 400;
`;
