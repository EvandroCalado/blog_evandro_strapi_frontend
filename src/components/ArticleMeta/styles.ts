import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['7xl']};
    margin: auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    & > span > a {
      font-weight: bold;
      transition: ${theme.transitions.faster};
      margin-left: ${theme.spacings.sm};

      &:hover {
        text-decoration: underline;
      }
    }

    ${Title} {
      width: 100%;
      max-width: ${theme.frameSizes['4xl']};
      text-align: center;
    }
`}
`;

export const Image = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 40rem;
    object-fit: cover;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
  `}
`;

export const ArticleContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    justify-content: space-between;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
    }
`}
`;

export const DateContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: end;
    justify-content: center;
    gap: ${theme.spacings.lg};
`}
`;
