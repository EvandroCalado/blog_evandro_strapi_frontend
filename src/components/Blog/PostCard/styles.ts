import styled, { css } from 'styled-components';
import { Title } from '../../Shared/Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    flex-direction: column;
    padding: ${theme.spacings.lg} ${theme.spacings.md};

    & > a {
      width: 100%;
      height: 20rem;
      margin-bottom: ${theme.spacings.md};
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 100%;
    margin-bottom: ${theme.spacings.md};

    & img {
      object-fit: cover;
      border-radius: ${theme.spacings.sm};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.md};

    & > a {
      transition: ${theme.transitions.faster};

      &:hover {
        text-decoration: underline;
        filter: invert(.2);
      }
    }

    ${Title} {
      overflow: hidden;
      display: -webkit-box;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 2;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.xs} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 4;
`}
`;

export const DateContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
`;

export const Date = styled.span`
`;
