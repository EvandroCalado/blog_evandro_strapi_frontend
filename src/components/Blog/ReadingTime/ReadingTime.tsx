import { readingTime } from 'reading-time-estimator';
import { IconClock } from '../../../icons/icon-clock';
import * as Styled from './ReadingTime.styles';

export interface ReadingTimeProps {
  content: string;
}

export default function ReadingTime({ content }: ReadingTimeProps) {
  const result = readingTime(content, 200, 'pt-br');

  return (
    <Styled.Container>
      <Styled.Span>{IconClock}</Styled.Span>
      <Styled.Span>{result.text}</Styled.Span>
    </Styled.Container>
  );
}
