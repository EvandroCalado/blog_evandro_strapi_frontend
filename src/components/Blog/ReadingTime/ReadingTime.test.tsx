import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../../styles/render-theme';
import ReadingTime from './ReadingTime';

describe('<ReadingTime />', () => {
  it('should render with default values', () => {
    renderTheme(<ReadingTime content="test" />);
    expect(
      screen.getByText('menos de um minuto de leitura'),
    ).toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = renderTheme(<ReadingTime content="test" />);
    expect(container).toMatchSnapshot();
  });
});
