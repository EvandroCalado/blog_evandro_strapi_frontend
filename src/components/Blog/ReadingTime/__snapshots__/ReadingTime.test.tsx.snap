// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadingTime /> should render a snapshot 1`] = `
.c0 {
  display: flex;
  align-content: center;
  gap: 1rem;
}

.c0 svg {
  width: 2rem;
}

.c1 {
  display: flex;
  align-items: end;
  font-weight: 600;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      <svg
        class="icon icon-tabler icon-tabler-clock-filled"
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="{1.5}"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
          stroke="none"
        />
        <path
          d="M17 3.34a10 10 0 1 1 -14.995 8.984l-.005 -.324l.005 -.324a10 10 0 0 1 14.995 -8.336zm-5 2.66a1 1 0 0 0 -.993 .883l-.007 .117v5l.009 .131a1 1 0 0 0 .197 .477l.087 .1l3 3l.094 .082a1 1 0 0 0 1.226 0l.094 -.083l.083 -.094a1 1 0 0 0 0 -1.226l-.083 -.094l-2.707 -2.708v-4.585l-.007 -.117a1 1 0 0 0 -.993 -.883z"
          fill="currentColor"
          stroke-width="{0}"
        />
      </svg>
    </span>
    <span
      class="c1"
    >
      menos de um minuto de leitura
    </span>
  </div>
</div>
`;
